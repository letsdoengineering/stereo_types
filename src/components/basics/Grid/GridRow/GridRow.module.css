.grid-row {
    display: flex;
}

.grid-row--padded {
    padding: var(--standard-padding)
}
