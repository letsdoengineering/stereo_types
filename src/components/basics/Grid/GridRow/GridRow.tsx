import React from 'react'
import getClassNames from '../../../../utils/get-class-names'
import * as gridRowStyles from './GridRow.module.css'

type GridRowProps = React.HTMLAttributes<HTMLDivElement> & {
    /** optional: class-names */
    className?: string
    /** indicates whether or not the row container should be padded */
    isPadded?: boolean
    /** children always of type ReactNode */
    children: React.ReactNode
}

const GridRow: React.FC<GridRowProps> = ({
    className,
    children,
    isPadded = false,
}: GridRowProps) => {
    const classNames = getClassNames({
        defaultClasses: [`${gridRowStyles.gridRow}`],
        className,
        conditionalClasses: {
            [`${gridRowStyles.gridRowPadded}`]: isPadded,
        },
    })

    return <div className={classNames}>{children}</div>
}

export default GridRow
