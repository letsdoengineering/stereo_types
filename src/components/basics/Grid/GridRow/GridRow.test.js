import renderer from 'react-test-renderer'
import GridRow from './GridRow'
import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

describe('GridRow', () => {
  it('has not changed the JSX layout of the page from previous snapshot', () => {
    const tree = renderer.create(<GridRow>some row</GridRow>).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the page DOM that is accessible', async () => {
    const { container } = render(<GridRow>some row</GridRow>)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
