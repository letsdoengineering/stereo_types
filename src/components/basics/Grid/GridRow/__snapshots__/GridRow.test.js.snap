// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridRow has not changed the JSX layout of the page from previous snapshot 1`] = `
<div
  className="gridRow"
>
  some row
</div>
`;
